### ISSUE TITLE
ale_aos_command.py output truncated at 3072 bytes

### ISSUE THREAD

**Issue by Adze1502** [gmoisio/ansible-aos-stdlib#1]
Hi,

Thank you for the AOS libraries. While trying to automate a backup of our Alcatel Omniswitches, the command "show configuration snapshot all" is always truncated to max 3072 bytes. I can't get my finger on the issue. The playbook i run:

```
- name: This is a test for ale_aos_config module
  hosts: alcatel
  gather_facts: false
  connection: local
  roles:
    - aos
  tasks:
    - name: RUN SHOW COMMAND
      ale_aos_command:
        host: "{{ inventory_hostname }}"
        username: "{{ ansible_user }}"
        password: "{{ ansible_password }}"
        command: "show configuration snapshot all"
      register: command_output
      vars:
        ansible_python_interpreter: "/usr/bin/python3"

    - name: PRINT OUTPUT TO TERMINAL WINDOW
      debug:
        msg: "{{ command_output.output }}"

    - name: SAVE OUTPUT TO FILE
      copy:
        content: "{{ command_output.output }}"
        dest: "~/backups/{{ inventory_hostname }}.conf"
      when: command_output.output
```

Do you have any idea why all my output is truncated to 3072 bytes? Thnx!

**Comment by gmoisio** [comment]
Hi, I tried the "show configuration snapshot all" on a 877 lines OS6900
configuration without any issue. You are truncated in the output variable
or in the file?

Looking at your playbook, I will find better to run a command to copy the
boot.cfg or vcboot.cfg to an FTP server, rather than getting the text to
copy it into a local file.

Best Regards

Le lun. 2 mars 2020 à 16:29, Jeroen Louwes <notifications@github.com> a
écrit :

**Comment by gmoisio** [comment]
Trying some switches, it looks like it's an AOS 6 limitation. Your issue is visible when trying to get configuration from an AOS 6 OS6450 switch.
It confirms that it will be better to try to get the file with an FTP copy.
Regards

**Comment by gmoisio** [comment]
Hi, I found a way to workaround the issue on AOS6. I need to do some more tests to be sure that there will be no side effect on AOS8.
Regards

**Comment by Adze1502** [comment]
Thanks for the quick reply. Looking forward to the work-around. For your info, the system i tested with is a AOS6 device:

```
# show system
System:
  Description:  Alcatel-Lucent OS6855-14 6.4.4.743.R01
```

**Comment by gmoisio** [comment]
Hi, I published the release 1.0.0 with some updates and the new feature to get all the output of an AOS6 device.
You can try this new release with the new option "timing":

```

      ale_aos_command:
        host: "{{ inventory_hostname }}"
        username: "{{ ansible_user }}"
        password: "{{ ansible_password }}"
        timing: True
        command: "show configuration snapshot all"

```
Instead of searching for a pattern in the output to determine a command is done (by default, this is based on the network device's prompt), it will wait for two additional seconds once there is no new data available.

Please let me know if it solves your issue...

Regards

**Comment by Adze1502** [comment]
Works like a charm! Thank you for this and for your help! 👍 🥇
